### PR TITLE
Subclass Gtk.Box instead of Gtk.ListBox in Mail.MessageListBox

### PR DESCRIPTION
**Pull request by leolost2605**
This is being done in preparation for a gtk4 port, as `Gtk.ListBox` will be final in gtk4. Also move the `HeaderBar` from its own class here.

- [x] Change class
- [x] Move its scrolled window to `MessageListBox` from `MainWindow`
- [x] Move headerbar
- [x] Rename to MessageList

### CODE REVIEW

**Comment by leolost2605** [comment]
Sorry, wanted to request review from the team, how do i do that? 😳

**Comment by lenemter** [comment]
@leolost2605 Only organization members can request review from the team

File: src/MessageList/MessageListItem.vala
Comment (lenemter) [inline src/MessageList/MessageListItem.vala:1]: Why `parent_message_list_box` was introduced here? Is accessing parent somehow broken in GTK 4 version?

File: src/MessageList/MessageListItem.vala [hunk 1]
```diff
@@ -354,8 +357,7 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
     }
 
     private void add_inline_composer (ComposerWidget.Type composer_type) {
-        var message_list_box = (MessageListBox) get_parent ();
-        message_list_box.add_inline_composer.begin (composer_type, this);
+        parent_message_list_box.add_inline_composer.begin (composer_type, this);
```
[the remark this answers is not in the document]
Comment (leolost2605) [reply src/MessageList/MessageListItem.vala]: Oh nice, didn't know about that, thank you!

**Comment by danirabbit** [comment]
@leolost2605 regarding the question in the PR description, I agree that moving the headerbar here and renaming this class to "MessageList" makes sense :+1:

File: src/MessageList/MessageListBox.vala [hunk 2]
```diff
@@ -18,12 +18,18 @@
...
     public GenericArray<string> uids { get; private set; default = new GenericArray<string> (); }
 
+    private Gtk.ListBox list_box;
+    private Gtk.ScrolledWindow scrolled_window;
+
     public MessageListBox () {
-        Object (selection_mode: Gtk.SelectionMode.NONE);
+        Object (
+            orientation: Gtk.Orientation.VERTICAL,
+            spacing: 0
```
Comment (danirabbit) [inline src/MessageList/MessageListBox.vala:31]: I think spacing is 0 by default so I'm not sure you need to set this explicitly

**Review by danirabbit — CHANGES_REQUESTED** [review]
One small inline suggestion. Also don't forget to update po/POTFILES :)

**Review by danirabbit — APPROVED** [review]

**Pull request merged**